src/tests/Microsoft.PowerFx.Connectors.Tests/OpenApiJsonSerializerTests.cs

```diff
@@ -0,0 +1,144 @@
+            Assert.Equal("Missing valid FormulaValue for property a", ex.Message);
+        }
+
+        [Fact]
+        public void JsonSerializer_Integer_String_Mismatch()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "integer" } } };
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.New("abc")) };
+
+            var ex = Assert.Throws<ArgumentException>(() => jsonSerializer.Serialize(schema, fields));
+            Assert.Equal("Type mismatch for property a, expected type Number and got String", ex.Message);
+        }
+
+        [Fact]
+        public void JsonSerializer_TwoIntegers()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "integer" }, ["b"] = new OpenApiSchema() { Type = "integer" } } };
+            var fields = new List<NamedValue>() { new NamedValue("b", FormulaValue.New(-2)), new NamedValue("a", FormulaValue.New(1)) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal(@"{""a"":1,""b"":-2}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_String()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "string" } } };
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.New("abc")) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal(@"{""a"":""abc""}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_Bool()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "boolean" }, ["b"] = new OpenApiSchema() { Type = "boolean" } } };
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.New(true)), new NamedValue("b", FormulaValue.New(false)) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal(@"{""a"":true,""b"":false}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_Array()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "array", Format = "int" } } };                            
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.NewSingleColumnTable(FormulaValue.New(1), FormulaValue.New(2))) };
```

Comment:
>NewSingleColumnTable [](http://example.com/codeflow?start=83&length=20)

This is good! Showing that a single-column-table serializes as an array (and no extra "Value" field shows up)